sdks/java/extensions/sketching/src/test/java/org/apache/beam/sdk/extensions/sketching/ApproximateDistinctTest.java

```diff
@@ -0,0 +1,230 @@
+    PCollection<Long> results = tp
+            .apply("per key stream", Create.of(stream))
+            .apply("create keys", WithKeys.<Integer, Integer>of(1))
+            .apply("per key cardinality", ApproximateDistinct
+                    .<Integer, Integer>perKey()
+                    .withPrecision(p))
+            .apply("extract values", Values.<HyperLogLogPlus>create())
+            .apply("retrieve per key cardinality", ParDo.of(new RetrieveDistinct()));
+
+    PAssert.that("Verify Accuracy for cardinality per key", results)
+            .satisfies(new VerifyAccuracy(cardinality, expectedErr));
+    tp.run();
+  }
+
+  @Test
+  public void customObject() {
+    final int cardinality = 500;
+    final int p = 15;
+    final Double expectedErr = 1.04 / Math.sqrt(p);
+
+    Schema schema = SchemaBuilder.record("User").fields()
+            .requiredString("Pseudo")
+            .requiredInt("Age")
+            .endRecord();
+    List<GenericRecord> users = new ArrayList<>();
+    for (int i = 1; i <= cardinality; i++) {
+      GenericData.Record newRecord = new GenericData.Record(schema);
+      newRecord.put("Pseudo", "User" + i);
+      newRecord.put("Age", i);
+      users.add(newRecord);
+    }
+      PCollection<Long> results = tp
+              .apply("Create stream", Create.of(users).withCoder(AvroCoder.of(schema)))
+              .apply("Test custom object", ApproximateDistinct.<GenericRecord>globally()
+                      .withInputCoder(AvroCoder.of(schema))
+                      .withPrecision(p))
+              .apply("Retrieve cardinality of custom object", ParDo.of(new RetrieveDistinct()));
+
+      PAssert.that("Verify Accuracy for custom object", results)
+              .satisfies(new VerifyAccuracy(cardinality, expectedErr));
+      tp.run();
+  }
+
+  @Test
+  public void testCoder() throws Exception {
+    HyperLogLogPlus hllp = new HyperLogLogPlus(12, 18);
+    for (int i = 0; i < 10; i++) {
+      hllp.offer(i);
+    }
+    CoderProperties.<HyperLogLogPlus>coderDecodeEncodeEqual(
+            ApproximateDistinct.HyperLogLogPlusCoder.of(), hllp);
+  }
+
+  @Test
+  public void testDisplayData() {
+    final ApproximateDistinctFn<Integer> fnWithPrecision =
+            ApproximateDistinctFn.create(BigEndianIntegerCoder.of()).withPrecision(23);
+
+    assertThat(DisplayData.from(fnWithPrecision), hasDisplayItem("inputCoder",
+            BigEndianIntegerCoder.of().getEncodedTypeDescriptor().toString()));
+    assertThat(DisplayData.from(fnWithPrecision), hasDisplayItem("p", 23));
+    assertThat(DisplayData.from(fnWithPrecision), hasDisplayItem("sp", 0));
+
+  }
+
+  static class RetrieveDistinct extends DoFn<HyperLogLogPlus, Long> {
+    @ProcessElement
+    public void apply(ProcessContext c) {
+      Long card = c.element().cardinality();
+      LOG.debug("Number of distinct Elements : " + card);
+      c.output(card);
+    }
+  }
+
+  class VerifyAccuracy implements SerializableFunction<Iterable<Long>, Void> {
```

Comment:
Can you use this function in all of the tests? Seems that in some of the tests its code is basically duplicated.